prow/hook/server.go

```diff
@@ -104,49 +121,157 @@ func (s *Server) demuxEvent(eventType, eventGUID string, payload []byte) error {
 	} else {
 		counter.Inc()
 	}
+	var srcRepo string
 	switch eventType {
 	case "issues":
 		var i github.IssueEvent
 		if err := json.Unmarshal(payload, &i); err != nil {
 			return err
 		}
+		srcRepo = i.Repo.FullName
 		go s.handleIssueEvent(l, i)
 	case "issue_comment":
 		var ic github.IssueCommentEvent
 		if err := json.Unmarshal(payload, &ic); err != nil {
 			return err
 		}
+		srcRepo = ic.Repo.FullName
 		go s.handleIssueCommentEvent(l, ic)
 	case "pull_request":
 		var pr github.PullRequestEvent
 		if err := json.Unmarshal(payload, &pr); err != nil {
 			return err
 		}
+		srcRepo = pr.Repo.FullName
 		go s.handlePullRequestEvent(l, pr)
 	case "pull_request_review":
 		var re github.ReviewEvent
 		if err := json.Unmarshal(payload, &re); err != nil {
 			return err
 		}
+		srcRepo = re.Repo.FullName
 		go s.handleReviewEvent(l, re)
 	case "pull_request_review_comment":
 		var rce github.ReviewCommentEvent
 		if err := json.Unmarshal(payload, &rce); err != nil {
 			return err
 		}
+		srcRepo = rce.Repo.FullName
 		go s.handleReviewCommentEvent(l, rce)
 	case "push":
 		var pe github.PushEvent
 		if err := json.Unmarshal(payload, &pe); err != nil {
 			return err
 		}
+		srcRepo = pe.Repo.FullName
 		go s.handlePushEvent(l, pe)
 	case "status":
 		var se github.StatusEvent
 		if err := json.Unmarshal(payload, &se); err != nil {
 			return err
 		}
+		srcRepo = se.Repo.FullName
 		go s.handleStatusEvent(l, se)
 	}
+	// Demux events only to external plugins that require this event.
+	if external := s.needDemux(eventType, srcRepo); len(external) > 0 {
+		go s.demuxExternal(l, external, payload, h)
+	}
+	return nil
+}
+
+// needDemux returns whether there are any external plugins that need to
+// get the present event.
+func (s *Server) needDemux(eventType, srcRepo string) []plugins.ExternalPlugin {
+	var matching []plugins.ExternalPlugin
+
+	for repo, plugins := range s.Plugins.Config().ExternalPlugins {
+		// Make sure the repositories match
+		var matchesRepo bool
+		if repo == srcRepo {
+			matchesRepo = true
+		}
+		// If this is an org, we need to compare orgs.
+		if !matchesRepo && !strings.Contains(repo, "/") {
+			parts := strings.Split(srcRepo, "/")
+			if len(parts) != 2 {
+				continue
+			}
+			if parts[0] == repo {
```

Comment:
nit: You can parse the org from `srcRepo` once outside the loop instead of for every `repo`.

/lgtm
/hold

Reply:
done